registry/registry.go

```diff
@@ -723,11 +738,70 @@ func tarReader(reader io.Reader, contentType string) (*tar.Reader, error) {
 	return tar.NewReader(reader), nil
 }
 
-func downloadVersion(opts *VersionOptions) (ver *Version, attachments []*kivik.Attachment, err error) {
-	url := opts.URL
+// CheckVersion controls the matching versions between retrieved tarball and
+// options
+func (t *Tarball) CheckVersion(expectedVersion string) (bool, error) {
+	tbManifestVersion := t.Manifest.Version
+	tbPackageVersion := t.PackageVersion
+
+	var match bool
+	var errm error
+
+	if tbManifestVersion != "" {
+		match = VersionMatch(expectedVersion, tbManifestVersion)
+	}
+	if !match {
+		errm = multierror.Append(errm,
+			fmt.Errorf("%q field does not match (%q != %q)",
+				"version", expectedVersion, tbManifestVersion))
+	}
+
+	if tbPackageVersion != "" {
+		match = VersionMatch(expectedVersion, tbPackageVersion)
+		if !match {
+			errm = multierror.Append(errm,
+				fmt.Errorf("\"version\" from package.json (%q != %q)",
+					expectedVersion, tbPackageVersion))
+		}
+	}
+
+	if errm != nil {
+		err := errshttp.NewError(http.StatusUnprocessableEntity,
+			"Content of the manifest does not match: %s", errm)
+		return false, err
+	}
+
+	return true, nil
+}
+
+// CheckEditor validates a tarball manifest editor
+func (t *Tarball) CheckEditor() (bool, error) {
+	editorName := t.Manifest.Editor
+
+	if editorName == "" {
+		return false, fmt.Errorf("%q field is empty", "editor")
```

Comment:
Why do you use a constant string in the second parameter? For example, this code looks less weird:

```go
return false, errors.New(`The "editor" field is empty`)
```